txpool/txpool.go

```diff
@@ -12,15 +12,29 @@ import (
 
 // A Pool holds transactions that may be included in future blocks.
 type Pool struct {
-	txns map[sunyata.TransactionID]sunyata.Transaction
-	vc   consensus.ValidationContext
-	mu   sync.Mutex
+	txns       map[sunyata.TransactionID]sunyata.Transaction
+	lastUpdate consensus.StateApplyUpdate
+	prevVC     consensus.ValidationContext
+	vc         consensus.ValidationContext
+	mu         sync.Mutex
 }
 
 func (p *Pool) validateTransaction(txn sunyata.Transaction) error {
 	// perform standard validation checks
 	if err := p.vc.ValidateTransaction(txn); err != nil {
-		return err
+		// If the transaction is valid against the previous validation context
+		// update the proofs and revalidate.
+		if prevErr := p.prevVC.ValidateTransaction(txn); prevErr != nil {
+			return err
+		}
```

Comment:
it's unusual to check `prevErr` but return `err`, so this should have a comment:
```go
return err // return the original error
```